parsec/core/gui/new_version.py

```diff
@@ -28,18 +26,49 @@ def _extract_version_tuple(raw):
         return None
 
 
-async def _do_check_new_version(url):
-    # urlopen automatically follows redirections
-    resolved_url = await trio.to_thread.run_sync(
-        lambda: urlopen(Request(url, method="HEAD")).geturl()
-    )
+def _tuple_to_version_number(version):
+    return ".".join([str(i) for i in version])
+
+
+async def _async_get(url, method="GET"):
+    return await trio.to_thread.run_sync(lambda: urlopen(Request(url, method=method)))
```

Comment:
We should consider an actual async library for http requests at some point.

Reply:
Yup, we should...